src/components/SchemaTree.tsx

```diff
@@ -16,18 +17,37 @@ export interface ISchemaTree extends IExtendableRenderers {
   expanded?: boolean;
   maxRows?: number;
   onGoToRef?: GoToRefHandler;
+  rowRenderer?: RowRenderer;
 }
 
 const canDrag = () => false;
 
 export const SchemaTree = observer<ISchemaTree>(props => {
-  const { hideTopBar, name, treeStore, maxRows, className, onGoToRef } = props;
+  const { hideTopBar, name, treeStore, maxRows, className, onGoToRef, rowRenderer: customRowRenderer } = props;
+
+  React.useEffect(
```

Comment:
I was thinking the other day on our code coverage in our React repos. It's tough to test React I agree but here's a though: why don't we embrace a pattern where we encapsulate all 'logic' before the `return`/`render` function inside external controller?

The idea is that you would move those effects in a separate component / function which could then be unit tested very easily.